controller/linodemachine_controller_helpers.go

```diff
@@ -74,8 +80,24 @@ func (r *LinodeMachineReconciler) newCreateConfig(ctx context.Context, machineSc
 
 		return nil, err
 	}
-	createConfig.Metadata = &linodego.InstanceMetadataOptions{
-		UserData: b64.StdEncoding.EncodeToString(bootstrapData),
+
+	region, err := machineScope.LinodeClient.GetRegion(ctx, machineScope.LinodeMachine.Spec.Region)
+	if err != nil {
+		return nil, err
+	}
+	regionMetadataSupport := slices.Contains(region.Capabilities, "Metadata")
+	if machineScope.LinodeMachine.Spec.UseStackScriptBootstrap || !regionMetadataSupport {
```

Comment:
We should also to check that the image also support the Metadata service here. Couldn't we check the `capabilities` in the Images List response? https://www.linode.com/docs/api/images/#images-list__response-samples

I see that this is implicitly called it out in the documentation changes (only Ubuntu is supported for now), but for future-proofing, this check should be made and failures (via log or error) communicated to the user to inform them of an invalid configuration.

Reply:
I did consider doing that at first, a few things I ran into looking into this. Image list/get is not cached by the linode client like the region endpoint is so we would have to make an extra call out to the linode API every time a new machine came up. additionally, even if we did make this call, private user images capabilities do not always reflect whether or not the image actually supports cloud init, they have to set it as a capability at image creation time and cannot update it after the image is created. If you still think it is worth it I can look into adding it in as a default, but I think we should still support overwriting the default

Reply:
> Image list/get is not cached by the linode client

I don't think we can use the caching provided by the `linodego` client in our current implementation anyways since we create a new client on every reconciliation loop.

> private user images capabilities do not always reflect whether or not the image actually supports cloud init, they have to set it as a capability at image creation time and cannot update it after the image is created.

Could we skip the `capabilities` check for private images then?

Reply:
we can check if they happen to have it enabled whether it is public/private and then just let them override it with this flag if it was incorrectly tagged?

Reply:
or we could limit the check to just `linode/` images

Reply:
Actually, thinking more about it, is the core issue here just user misconfiguration? Would this validation be better served in a admission webhook?